syz-fuzzer/proc.go

```diff
@@ -201,12 +216,17 @@ func getSignalAndCover(p *prog.Prog, info *ipc.ProgInfo, call int) (signal.Signa
 }
 
 func (proc *Proc) smashInput(item *WorkSmash) {
+	// If MAB seed scheduling enabled, only do this for first-timers.
 	if proc.fuzzer.faultInjectionEnabled && item.call != -1 {
 		proc.failCall(item.p, item.call)
 	}
 	if proc.fuzzer.comparisonTracingEnabled && item.call != -1 {
 		proc.executeHintSeed(item.p, item.call)
 	}
+	// If MAB seed scheduling enabled, do not mutate for 100 times.
```

Comment:
Yes, in the research we disabled all smash mutations when MAB is in play. In practice, the 100 forced smashes will cause an extremely long smash-triage-smash chain reaction, which is actually detrimental to corpus diversity. With MAB, old seeds are very likely to have negative weight and new seeds will still be mutated.

Of course, it can be argued that we could reduce the number of smashes performed. In the research, we went for the extreme to demonstrate MAB can do it just as well. 

Also, I realize that if we do want to disable smash mutations completely, the fault inject logic will also need to be moved as well.

Reply:
I had an idea that might get us the best of both worlds: Run the smash mutations for up to 100 times and compute their reward as if they were chosen with a certain probability (e.g. 100%). Break the smash loop as soon as the total seed reward drops below a certain threshold (e.g. 0). This way we can thoroughly test the new seeds first (which can be slow for MAB alone when there are thousands of seeds) while abandoning bad seeds quickly.

Needs further testing.

Reply:
> Break the smash loop as soon as the total seed reward drops below a certain threshold (e.g. 0).

This sounds reasonable.
But one way or another dead code is not good, especially in such covert form. The code that never runs should be removed.